Side by side host config (#5605)
hypothesis/client: docs/publishers/config.rst

                              // relative from the receiving iframe.
    }

.. option:: sideBySide

>   ``Object``. This option lets you customize how the side-by-side mode behaves.

Reviewer: I think it would be useful to explain briefly what "side-by-side mode" means here.

  For example:

  .. code-block:: javascript

>      window.hypothesisConfig = function () {

Reviewer: We should probably start using modern JS (ie. arrow functions) in our code examples now.

Author: Right! I basically copy-pasted another example 😅

       return {
         sideBySide: {
           mode: 'manual'
         }
       };
     };

  The following keys are supported in the :option:`sideBySide` object.

  .. option:: mode

    ``auto`` or ``manual``. Auto is the default value, where the main page content will be adjusted based on the sidebar

>     status, while manual indicates the guest app wants to take full control of handling side-by-side, which will

Reviewer: ```suggestion
    status, while manual indicates the web page wants to take full control of handling side-by-side, which will
```

The terms "guest" and "host" are used internally, but I don't think we use that terminology anywhere in our docs at present.

>     effectively disable that logic.

Reviewer: It would be useful to provide some pointers here about how the web page is supposed to implement its own side-by-side logic (ie. using the `hypothesis:layoutchange`) event.

Author: I was planning to link to that event's documentation, with a brief explanation, but I switch to something else and forgot.

                              // relative from the receiving iframe.
    }

.. option:: sideBySide

>   ``Object``. This option lets you customize how the side-by-side mode behaves.

Reviewer: ```suggestion
  ``Object``. This option lets you customize how side-by-side mode behaves.
```

  .. note::

>     The side-by-side mode allows the space used by the web page's main content

Reviewer: ```suggestion
    Side-by-side mode allows the space used by the web page's main content
```

    area to be adapted while the sidebar is open, ensuring it does not overlap

>     with potentially annotatable content.

Reviewer: ```suggestion
    with annotatable content.
```

"Potentially" seems superfluous here.

  For example:

  .. code-block:: javascript

     window.hypothesisConfig = () => ({
       sideBySide: {
         mode: 'manual'
       }
     });

  The following keys are supported in the :option:`sideBySide` object.

  .. option:: mode

    ``auto`` or ``manual``. Auto is the default value, where the main page

>     content will be adjusted based on the sidebar status, while manual

Reviewer: ```suggestion
    content will be automatically resized to fit alongside the sidebar, while manual
```

The meaning of "status" might be unclear here.

>     indicates the web page wants to take full control of handling side-by-side,
>     which will effectively disable that logic.

Reviewer: ```suggestion
    indicates the web page wants to take full control of handling side-by-side.
    This disables automatic resizing of the content.
```

I'm being more explicit about what "that logic" refers to.

    When setting it to ``manual``, you probably want to
    `listen for layout changes </publishers/events/#cmdoption-arg-hypothesis-layoutchange>`_

>     as well, so that you can adapt the main content area of the page.

Reviewer: I think we should re-word this to indicate that web page authors _should_ respond to layout changes and adapt their content if that is reasonably possible, so as to provide an experience that is consistent with how Hypothesis works more generally. There will be cases where it is not reasonable, in which case the page author can just set the mode to "manual" and do nothing else.

Possible wording: "When setting this to `manual`, you should listen for layout changes and adapt content to fit alongside the sidebar, if it is reasonable to do so."

   the URL where the contents of the hypothesis package are hosted, including
   the trailing slash. (Default: for production builds:
   ``"https://cdn.hypothes.is/hypothesis/X.Y.Z/"``, for development builds:

>    ``"http://localhost:3001/hypothesis/X.Y.Z/"``.

Author: Small unrelated typo fix.
